examples/shopify/lib/config.ts

```diff
@@ -1,30 +1,58 @@
-export type Config = {
+type PublicConfig = {
   shopify: {
     storeName: string
     accessToken: string
   }
+}
+
+type PrivateConfig = {
   makeswift: {
     siteApiKey: string
     productTemplatePathname: string
   }
 }
 
-function getEnvVarOrThrow(key: string): string {
-  const value = process.env[key]
+export type Config = PublicConfig & PrivateConfig
 
+/**
+ * public env values must be inlined for the value to be exposed to the browser
+ * https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser
+ * @param key -> name of env variable
+ * @param value -> value option of env variable
+ * @returns value
+ */
+function verifyPublicEnvValueOrThrow(key: string, value?: string): string {
   if (!value) throw new Error(`"${key}" env var is not defined.`)
 
   return value
 }
 
-export function getConfig(): Config {
+function getPrivateEnvValueOrThrow(key: string) {
+  const value = process.env[key]
+
+  return verifyPublicEnvValueOrThrow(key, value)
+}
+
+export function getPublicConfig(): PublicConfig {
   return {
     shopify: {
-      accessToken: getEnvVarOrThrow('SHOPIFY_ACCESS_TOKEN'),
-      storeName: getEnvVarOrThrow('SHOPIFY_STORE_NAME'),
+      accessToken: verifyPublicEnvValueOrThrow(
+        'NEXT_PUBLIC_SHOPIFY_ACCESS_TOKEN',
```

Comment:
The key here doesn't seem to be used except for writing the error message. This functionality is a very generic "make sure this value exists, otherwise throw". This is commonly known as an "invariant", popularize by Facebook OSS libraries. There's a few variants of it out there, this one being the most popular (I think): https://github.com/alexreardon/tiny-invariant

Reply:
> The key here doesn't seem to be used except for writing the error message

True. This is because Next.js wont include the env in the browser unless I write out `process.env.actualname` but I also want the variable name for error handling so I am passing in both.

https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

Reply:
Thank you for explaining that BTW. I had seen that before and been confused. Finally connected the dots by reading your comment and reading that package readme